### ISSUE TITLE
Sample limit order book export to csv...

### ISSUE THREAD

**Issue by DanielRonnstam** [phil8192/limit-order-book#2]
Hello. For research purposes, I need a script that can parse a message flow of Level II limit order data (order, delete, modify, trades etc), and sample the order book to a CSV for each ticker symbol (stock) in the following format;

- Date.
- Day of week.
- Timestamp, milliseconds from midnight (interval to be set in script command, 1 sec to 10 min interval).
- bidprice_level_1
- bidvolume_level_1
- askprice_level_1
- askvolume_level_1 
- bidprice_level_2
- etc.. to level 20.

The messages are converted from Nasdaq ITCH 5.0 and are loaded into an R data.table, or exported to CSV. I have this already from another script.

Can your script do this sampling? Or can it be easily modified to do the sampling? Or, do you know some other script that can? thanks

**Comment by RJ2000** [comment]
How are you getting TotalView ITCH data? Who’s is the data provider?

**Comment by DanielRonnstam** [comment]
I downloaded the file from nasdaqs demo server. It contains level 2 data for all instruments in the exchanges the cover, during one day. I will use this data for testing the script.

**Comment by RJ2000** [comment]
I don’t think you have enough data if you only got the snapshot. It’s easy to transform the data by timestamp.

**Comment by RJ2000** [comment]
where is the link to the sample data?

**Comment by phil8192** [comment]
Hi @DanielRonnstam, It has been a while since I've done anything with this code - There is likely something much nicer out there that can do what you want. I am however planning to make some changes so that it can take as input a CSV containing limit order events and output a CSV with summary stats for each LOB update.

Sticking to basic limit order types, the input CSV might look like:

```
order_id, local_timestamp, exchange_timestamp, price, volume, action, direction
```

Where `order_id` = Unique limit order id, `local_timestamp` (optional) time limit order event recorded, `exchange_timestamp` = time of limit order event, `price` = limit order price, `volume` = limit order volume, `action` = one of `added`, `deleted`, `updated` (update may be for price and/or volume), `direction` = Bid or Ask.

**Comment by RJ2000** [comment]
That would be great @phil8192 are you able to get LOB data for stocks?

**Comment by DanielRonnstam** [comment]
Thanks. With LOB update, what do you mean? Do you know of any other order book sampling script that does what I need it to do?
